src/jinja2/filters.py

```diff
@@ -772,12 +833,17 @@ def do_indent(
 
     newline = "\n"
 
-    if isinstance(s, Markup):
-        indention = Markup(indention)
-        newline = Markup(newline)
+    if hasattr(s, "__html__"):
+        indention = eval_ctx.mark_safe(indention)
+        newline = eval_ctx.mark_safe(newline)
+        # Make sure the correct MarkupClass is used
```

Comment:
```suggestion
        # Make sure the correct Markup class is used
```

At least in other places it is written like that.